craft_application/application.py

```diff
@@ -178,19 +184,73 @@ def _configure_services(
             work_dir=self._work_dir,
         )
 
-    @functools.cached_property
-    def project(self) -> models.Project:
-        """Get this application's Project metadata."""
-        # Current working directory contains the project file
-        project_file = pathlib.Path(f"{self.app.name}.yaml").resolve()
-        craft_cli.emit.debug(f"Loading project file '{project_file!s}'")
+    def _resolve_project_path(self, project_dir: pathlib.Path | None) -> pathlib.Path:
+        """Find the project file for the current project.
+
+        The default implementation simply looks for the project file in the project
+        directory. Applications may wish to override this if the project file could be
+         in multiple places within the project directory.
+        """
+        if project_dir is None:
+            project_dir = pathlib.Path.cwd()
+        return (project_dir / f"{self.app.name}.yaml").resolve(strict=True)
```

Comment:
The `strict=True` is also a change, right? Is there a specific reason for it?

Reply:
It bumps the same FileNotFoundError up earlier. Rather than waiting until we try to open it, it fails when resolving the path. No other real reason, so I'd happily undo it, but I don't think it changes the behaviour.